src/BloomBrowserUI/bookEdit/js/calledByCSharp.ts

```diff
@@ -2,20 +2,44 @@
 
 class CalledByCSharp {
 
-  handleUndo(): string {
+  handleUndo(): void {
+    // Stuff "in the accordion" (not clear what that means) gets its own undo handling
     var contentWindow = this.getAccordionContent();
-    if (!contentWindow || !contentWindow.model || !contentWindow.model.shouldHandleUndo()) {
-      return 'fail';
+    if (contentWindow && contentWindow.model && contentWindow.model.shouldHandleUndo()) {
+          contentWindow.model.undo();
+    } // elsewhere, we try to ask ckEditor to undo, else just the document
+    else{
+        var ckEditorUndo = this.ckEditorUndoCommand();
+        if (ckEditorUndo === null || !ckEditorUndo.exec()) {
+            //sometimes ckEditor isn't active, so it wasn't paying attention, so it can't do the undo. So ask the document to do an undo:
+            (<any>this.getPageContent()).document.execCommand('undo', false, null);
+        }
     }
-    contentWindow.model.undo();
-    return 'success';
   }
 
+ ckEditorUndoCommand(): any {
+     try {
+         return (<any>this.getPageContent()).CKEDITOR.instances.editor1.commands.undo;
+     }
+     catch (e) {
+         return null;
+     }
+ }
+
   canUndo(): string {
     var contentWindow = this.getAccordionContent();
-    if (!contentWindow || !contentWindow.model || !contentWindow.model.shouldHandleUndo())
-      return 'fail'; // we don't want to decide
-    return contentWindow.model.canUndo();
+    if (contentWindow && contentWindow.model && contentWindow.model.shouldHandleUndo()) {
+          return contentWindow.model.canUndo();
+    }
+    /* I couldn't find a way to ask ckeditor if it is ready to do an undo.
+      The "canUndo()" is misleading; what it appears to mean is, can this command (undo) be undone?*/
+
+    /*  var ckEditorUndo = this.ckEditorUndoCommand();
+        if (ckEditorUndo === null) return 'fail';
+        return ckEditorUndo.canUndo() ? 'yes' : 'no';
+    */
+
+      return "fail"; //go ask the browser
```

Comment:
So this particular method didn't really change functionally, but has some sample code that doesn't work left in to keep others from going down that rabbit trail.